### PR TITLE
Check pointing coordinate frame in DispReconstructor

### LINKED ISSUE #2427: DispRegressor doesn't check in which coordinate system the array pointing is given
**Describe the bug**

DispRegressor reads the subarray pointing from the observation block data, assumes it's alt az, but for observed data like LST-1, this will not be true, as the tracking is constant in ICRS (ra/dec), not horizontal coordinates.

### PR DESCRIPTION
**Pull request by LukasBeiske**
Fixes #2427.

### CODE REVIEW

**Review by Tobychev — APPROVED** [review]

File: ctapipe/reco/sklearn.py [hunk 1]
```diff
@@ -777,13 +778,34 @@ def predict_table(self, key, table: Table) -> Dict[ReconstructionProperty, Table
         fov_lon = table["hillas_fov_lon"].quantity + disp * np.cos(psi)
         fov_lat = table["hillas_fov_lat"].quantity + disp * np.sin(psi)
 
-        # FIXME: Assume constant and parallel pointing for each run
-        self.log.warning("Assuming constant and parallel pointing for each run")
+        self.log.warning("FIXME: Assuming constant and parallel pointing for each run")
+        if np.all(table["subarray_pointing_frame"] is CoordinateFrameType.ALTAZ):
```
Comment (maxnoe) [inline ctapipe/reco/sklearn.py:782]: this won't work. `is` cannot be applied element wise and these checks will always be False.

You need to check `np.all(table["subarray_pointing_frame"] == CoordinateFrameType.ALTAZ.value)`

File: ctapipe/reco/sklearn.py [hunk 2]
```diff
@@ -777,13 +778,34 @@ def predict_table(self, key, table: Table) -> Dict[ReconstructionProperty, Table
...
+            pointing_alt = table["subarray_pointing_lat"]
+            pointing_az = table["subarray_pointing_lon"]
+        elif np.all(table["subarray_pointing_frame"] is CoordinateFrameType.ICRS):
+            pointing_altaz = SkyCoord(
+                ra=table["subarray_pointing_lon"],
+                dec=table["subarray_pointing_lat"],
+                frame="icrs",
+            ).transform_to(AltAz())
```
Comment (maxnoe) [inline ctapipe/reco/sklearn.py:790]: This also won't work. Transformation from ICRS to the AltAz system needs location and observation time

File: ctapipe/reco/sklearn.py [hunk 3]
```diff
@@ -777,13 +778,34 @@ def predict_table(self, key, table: Table) -> Dict[ReconstructionProperty, Table
...
+            pointing_alt = pointing_altaz.alt
+            pointing_az = pointing_altaz.az
+        elif np.all(table["subarray_pointing_frame"] is CoordinateFrameType.GALACTIC):
+            pointing_altaz = SkyCoord(
+                l=table["subarray_pointing_lon"],
+                b=table["subarray_pointing_lat"],
+                frame="galactic",
+            ).transform_to(AltAz())
```
Comment (maxnoe) [inline ctapipe/reco/sklearn.py:798]: Same as above, needs obstime and location

**Review by maxnoe — CHANGES_REQUESTED** [review]

**Comment by maxnoe** [comment]
I think this is partially already addressed (for the application  case) in #2409 and you should just throw an error if the training data is not fixed altaz

**Review by Tobychev — APPROVED** [review]

**Pull request merged**